### PR TITLE
feat(trusted.ci.jenkins.io) Fine tune the security groups to restrict everything except explicit rules

### PR DESCRIPTION
**Pull request by dduportal**
Related to https://github.com/jenkins-infra/helpdesk/issues/3486

This PR fine tune the security groups to:

- Forbid everything by default
- Allow Bounce to reach the controller through HTTPS and TCP Inbound protocol
- Allow VMs to reach Internet through HTTP and HTTPS
- Allow VMs to reach the puppet master though the puppet protocol

### CODE REVIEW

File: trusted.ci.jenkins.io.tf [hunk 1]
```diff
@@ -503,3 +553,66 @@ resource "azurerm_network_security_rule" "allow_inbound_ssh_from_bounce_to_ephem
...
+  source_address_prefix       = "Internet"
+  destination_address_prefix  = "*"
+  resource_group_name         = data.azurerm_resource_group.trusted.name
+  network_security_group_name = azurerm_network_security_group.trusted_ci_controller.name
+}
+
+# This rule overrides an Azure-Default rule. its priority must be < 65000
+resource "azurerm_network_security_rule" "deny_all_inbound_from_vnet" {
+  name                        = "deny-all-inbound-from-vnet"
+  priority                    = 4096 # Maximum value allowed by the
```
Comment (lemeurherve) [inline trusted.ci.jenkins.io.tf:608]: "by the" what?

**Review by lemeurherve — APPROVED** [review]

File: trusted.ci.jenkins.io.tf (hunk 1 above)
Comment (lemeurherve) [reply trusted.ci.jenkins.io.tf:608]: Erf, automerged...

Comment (dduportal) [reply trusted.ci.jenkins.io.tf:608]: I'll fix it on a subsequent PR, no worries

Comment (dduportal) [reply trusted.ci.jenkins.io.tf:608]: Fixed in https://github.com/jenkins-infra/azure/pull/357

**Pull request merged**